test/e2e/test/elasticsearch/steps_creation.go

```diff
@@ -32,8 +32,9 @@ func (b Builder) CreationTestSteps(k *test.K8sClient) test.StepList {
 					if b.Elasticsearch.Spec.Version != createdEs.Spec.Version {
 						return fmt.Errorf("expected version %s but got %s", b.Elasticsearch.Spec.Version, createdEs.Spec.Version)
 					}
-					// TODO this is incomplete
-					return nil
+					// check cluster health here already to make sure any dependent builders can rely on a ready ES
+					// this is special casing Elasticsearch a little bit in the interest of test stability see https://github.com/elastic/cloud-on-k8s/issues/7172
+					return clusterHealthIs(b, k, esv1.ElasticsearchGreenHealth)
```

Comment:
nit: we already have a dedicated step, no real advantage compared to your proposal except that we would have a dedicated entry in the logs:

```
=== RUN   TestAgentConfigRef/Elasticsearch_cluster_should_be_created
Retries (5m0s timeout): .
    --- PASS: TestAgentConfigRef/Elasticsearch_cluster_should_be_created (0.57s)
=== RUN   TestAgentConfigRef/ES_cluster_health_should_eventually_be_green
Retries (5m0s timeout): ......
    --- PASS: TestAgentConfigRef/ES_cluster_health_should_eventually_be_green (41.40s)
=== RUN   TestAgentConfigRef/Creating_an_Agent_should_succeed
    --- PASS: TestAgentConfigRef/Creating_an_Agent_should_succeed (1.48s)
```

```suggestion
					return nil
				}),
			},
			// check cluster health here already to make sure any dependent builders can rely on a ready ES
			// this is special casing Elasticsearch a little bit in the interest of test stability see https://github.com/elastic/cloud-on-k8s/issues/7172
			CheckClusterHealth(b, k),
```

Reply:
I think my motivation was to have this check in the create steps. While the existing check runs in the check steps. By having it in the create steps I want to introduce a delay so that the Agent creation (and other resource creations) does not run before ES is ready. I could remove the dedicated step from the check steps to avoid the duplication.

Reply:
Ah I misunderstood your proposal!